src/openms/source/FORMAT/FileTypes.cpp

```diff
@@ -160,6 +162,27 @@ namespace OpenMS
     return r == FileTypes::Type::UNKNOWN ? fallback : r;
   }
 
+  static FileTypeList typesWithProperties(std::unordered_set<FileTypes::FileProperties> haveFeatures)
+  {
+    std::vector<FileTypes::Type> returnList;
+    std::vector<TypeNameBinding> goodTypes;
+    // Copy our type_with_annotation__s to a vector
+    std::copy(type_with_annotation__.begin(), type_with_annotation__.end(), std::back_inserter(goodTypes));
+    // for each feature we are looking for
+    for (auto i : haveFeatures)
+    {
+      // Remove any types that lack the feature
+      goodTypes.erase(std::remove_if(goodTypes.begin(), goodTypes.end(),[i](auto j) { return (j.features.find(i) != j.features.end()); }), goodTypes.end());
```

Comment:
Could probably be done without erasing … e.g. adding those that have features

Reply:
Currently, you are testing that a type needs to support all features.
You can iterate over all types, check with std::includes, if true add the type

on second thought it probably only works with set and not with unordered_set